backend/src/test/java/de/aljoschanyang/backend/services/TopicServiceTest.java

```diff
@@ -0,0 +1,53 @@
+package de.aljoschanyang.backend.services;
+
+import de.aljoschanyang.backend.models.Topic;
+import de.aljoschanyang.backend.models.TopicDTO;
+import de.aljoschanyang.backend.repositories.TopicRepo;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
+
+class TopicServiceTest {
+    private final TopicRepo mockTopicRepo = mock(TopicRepo.class);
+    private final TopicService topicService = new TopicService(mockTopicRepo);
+
+    @Test
+    void getAllTopics_whenNoTopicsInDB_thenReturnEmptyList() {
+        List<TopicDTO> expected = List.of();
+
+        when(topicService.getAllTopics()).thenReturn(expected);
```

Comment:
You want to test the service layer so it is more appropriate to mock the TopicRepo, as seen in the other test
```suggestion
        when(mockTopicRepo.findAll()).thenReturn(Collections.emptyList());
```

Reply:
I changed it